Src/Support/GoogleApis.Core/Apis/Requests/Parameters/ParameterUtils.cs

```diff
@@ -121,7 +125,24 @@ private static void IterateParameters(object request, Action<RequestParameterTyp
                 // Call action with the type name and value.
                 if (propertyType.IsValueType || value != null)
                 {
-                    action(attribute.Type, name, value);
+                    if (attribute.Type == RequestParameterType.UserDefiniedQueries)
+                    {
+                        if (value.GetType() == typeof(IEnumerable<KeyValuePair<string, string>>))
+                        {
+                            foreach (var pair in (IEnumerable<KeyValuePair<string, string>>)value)
+                            {
+                                action(RequestParameterType.Query, pair.Key, pair.Value);
+                            }
+                        }
+                        else
+                        {   
+                            //Do logging here

```

Comment:
@peleyal Once the logger can get set up, let me know what you'd like the logger to say and I'll add it in.